sync/fetch_queue.go

```diff
@@ -95,11 +94,8 @@ func (fq *fetchQueue) work() error {
 				fq.handleFetch(bjb)
 				fq.Info("next batch")
 			}
-			wg.Done()
 		}()
 	}
-	wg.Wait()
```

Comment:
I wonder how this did not cause stop test to fail - since it shuts down without making sure all goroutines exited - is this part of the fix?

Reply:
this waitgroup is totally ineffectual since this method is _always_ called in its own goroutine, and the caller is never waiting for it to return. removing it has no effect.

Reply:
the waitgroup postpones `shutdownRecover()` until after the workers have all finished, but I don't think that actually does anything. I don't think it would recover from a panic inside a child goroutine anyway.